lang-ui/src/main/java/at/tugraz/sw20asd/lang/ui/controllers/StudyVocab.java

```diff
@@ -0,0 +1,289 @@
+    @FXML
+    private Button switch_btn;
+    @FXML
+    private Button return_btn;
+    @FXML
+    private Button show_all_btn;
+    @FXML
+    private Label study_label;
+    @FXML
+    private AnchorPane anchor_pane;
+    @FXML
+    private Label user_info;
+    @FXML
+    private Label choice_label;
+    @FXML
+    private Label given_label;
+    @FXML
+    private ComboBox<String> language_choice;
+    @FXML
+    private Button start_btn;
+
+    private List<Button> answer_button_list = new ArrayList<>();
+    private List<Label> given_label_list = new ArrayList<>();
+    private List<Label> answer_label_list = new ArrayList<>();
+
+    private List<String> languages_list = new ArrayList<>();
+    private String language1;
+    private String language2;
+    private List<EntryDto> words = new ArrayList<>();
+
+    private Task<List<VocabularyDetailDto>> getVocabsTask;
+    private List<VocabularyDetailDto> vocabulary_list = new ArrayList<>();
+
+    private VocabularyAccess vocab;
+    private ObservableList<VocabularySelectionModel> voc;
+    private String study_language;
+    FXMLLoader loader = new FXMLLoader();
+
+    private String hide_all = "Hide all Answers";
+    private String show_all = "Show all Answers";
+    private String hide = "Hide Answer";
+    private String show = "Show Answer";
+
+    public StudyVocab(VocabularyAccess vocab, ObservableList<VocabularySelectionModel> voc) {
+        this.vocab = vocab;
+        this.voc = voc;
+        URL location = getClass().getResource("/study.fxml");
+        loader.setControllerFactory(c -> this);
+        loader.setRoot(this);
+        try {
+            loader.load(location.openStream());
+        } catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public void initialize() {
+        user_info.setVisible(false);
+        setStudyInterfaceVisible(false);
+        language1 = voc.get(0).getVocabularySrc();
+        language2 = voc.get(0).getVocabularyTarget();
+        languages_list.add(language1);
+        languages_list.add(language2);
+        ObservableList<String> languages = FXCollections.observableList(languages_list);
+        language_choice.setItems(languages);
+
+        start_btn.setOnAction(new EventHandler<ActionEvent>() {
+            public void handle(ActionEvent event) {
+                if (language_choice.getSelectionModel().isEmpty()) {
+                    user_info.setVisible(true);
+                    user_info.setText("Please choose a language!");
+                } else {
+                    study_language = language_choice.getSelectionModel().getSelectedItem();
+                    study_label.setText(study_language);
+                    getEntryList();
+                }
+            }
+        });
+
+        return_btn.setOnAction(new EventHandler<ActionEvent>() {
+            public void handle(ActionEvent event) {
+                OverviewVocabs overview = new OverviewVocabs(vocab);
```

Comment:
You should switch to Selection Overview, not Overview 

You do not delete/empty/stop everything (lists, tasks etc). This can cause major problems, especially if the task is not aborted.